### PR TITLE
feat: update keda to support autoscaling/v2

### PR DESCRIPTION
**Pull request by CeerDecy**
#### What this PR does / why we need it:
 update keda to support autoscaling/v2

#### Which issue(s) this PR fixes:

- Fixes #your-issue_number
- [Erda Cloud Issue Link](https://erda.cloud/erda/dop/projects/387/issues/all?id=533445&iterationID=12783&type=TASK)

#### Specified Reviewers:

/assign @iutx 

#### ChangeLog

| Language | Changelog |
| --------- | ------------ |
| 🇺🇸 English |       update keda to support autoscaling/v2        |
| 🇨🇳 中文    |       更新至Keda 2.14.0 适配 autoscaling/v2       |

#### Need cherry-pick to release versions?

Add comment like `/cherry-pick release/1.0` when this PR is merged.

> For details on the cherry pick process, see the [cherry pick requests](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md#how-to-cherry-pick-a-merged-pr) section under [CONTRIBUTING.md](https://github.com/erda-project/erda/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by iutx — COMMENTED** [review]
Could replace http client to clientgo？ @CeerDecy

**Review by iutx — DISMISSED** [review]
LGTM @sfwn

File: go.mod [hunk 1]
```diff
@@ -474,11 +471,20 @@ require (
...
 	sigs.k8s.io/cli-utils v0.35.0 // indirect
 	sigs.k8s.io/json v0.0.0-20221116044647-bc3834ca7abd // indirect
-	sigs.k8s.io/kustomize/api v0.13.5-0.20230601165947-6ce0bf390ce3 // indirect
-	sigs.k8s.io/kustomize/kyaml v0.14.3-0.20230601165947-6ce0bf390ce3 // indirect
+	sigs.k8s.io/kustomize/api v0.17.3 // indirect
+	sigs.k8s.io/kustomize/kyaml v0.17.2 // indirect
 	sigs.k8s.io/structured-merge-diff/v4 v4.4.1 // indirect
 )
 
+require (
```
Comment (sfwn) [inline go.mod:479]: merge into one require part.

File: internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8s.go [hunk 2]
```diff
@@ -1764,7 +1766,36 @@ func (k *Kubernetes) reApplyErdaHPARules(sg apistructs.ServiceGroup) (interface{
...
+		}, old)
+
+		if err != nil {
+			return nil, err
+		}
+
+		spec := types.ScaledPatchStruct{}
+
+		spec.Spec.ScaleTargetRef = scaledObj.Spec.ScaleTargetRef
+		if *scaledObj.Spec.MinReplicaCount > 0 && *scaledObj.Spec.MaxReplicaCount > 0 && *scaledObj.Spec.MinReplicaCount < *scaledObj.Spec.MaxReplicaCount {
```
Comment (sfwn) [inline internal/tools/orchestrator/scheduler/executor/plugins/k8s/k8s.go:1783]: Should check `scaleObj` is not nil before `*` at client side even if func `convertToKedaScaledObject` always return non-nil value.

**Review by sfwn — CHANGES_REQUESTED** [review]

**Review by sfwn — APPROVED** [review]

**Pull request merged**